Algebra/PartialOrd.hs

```diff
@@ -98,6 +99,11 @@ class Eq a => PartialOrd a where
 partialOrdEq :: PartialOrd a => a -> a -> Bool
 partialOrdEq x y = leq x y && leq y x
 
+instance PartialOrd () where
+    leq _ _ = True
+
+instance PartialOrd Void where
```

Comment:
also `leq _ _ = True` (that's how `==` is defined as well)

Reply:
Interesting... So, `leq (undefined :: Void) (undefined :: Void)` would panic, whereas the same for `(==)` wouldn't, right?

Reply:
somewhat so yes.

Reply:
Ah, I probably misinterpreted how Haskell handles empty instance bodies. I though there was some kind of totality checking involved that would find out that the `Void` cases aren't possible. Instead this would just emit an incomplete instance, which I didn't even know was possible.